tests-functional/tests/test_wakuext_community_token_permissions.py

```diff
@@ -0,0 +1,286 @@
+            {
+                "type": CommunityTokenType.ERC20,
+                "contract_addresses": {31337: self.snt_address},
+                "symbol": "SNT",
+                "amountInWei": "1000000000000000000",  # 1 token
+                "decimals": 18,
+            }
+        ]
+
+        # Add BECOME_MEMBER permission so join can succeed based on tokens
+        self.owner.wakuext_service.create_community_token_permission(
+            community_id=community_id,
+            permission_type=CommunityTokenPermissionType.BECOME_MEMBER,
+            token_criteria=token_criteria,
+        )
+
+        # Add BECOME_ADMIN permission with same criteria
+        self.owner.wakuext_service.create_community_token_permission(
+            community_id=community_id,
+            permission_type=CommunityTokenPermissionType.BECOME_ADMIN,
+            token_criteria=token_criteria,
+        )
+
+        # Fetch community as member
+        self.fetch_community(self.member_with_snt, community_id)
+
+        for i in range(10):
+            permissions_resp = self.member_with_snt.wakuext_service.check_permissions_to_join_community(community_id)
+            # Inspect response to see if admin permission is satisfied
+            # Exact response shape depends on your RPC wrapper; pseudocode:
+            if permissions_resp.get("satisfied"):
+                break
+            elif i == 9:
+                pytest.fail("Permissions to join never became satisfied for member_with_snt")
+
+            time.sleep(1)
+
+        # Member with tokens requests to join community
+        join_resp = self.member_with_snt.wakuext_service.request_to_join_community(community_id, member_address)
+        requests = join_resp.get("requestsToJoinCommunity", [])
+
+        if requests:
+            req_id = requests[0].get("id")
+            # Wait for token validation
+
+            time.sleep(2)
+            # Owner accepts the request since member has tokens
+            accept_resp = self.owner.wakuext_service.accept_request_to_join_community(req_id)
+            assert accept_resp is not None, f"Failed to accept request: {accept_resp}"
+
+        # Explicitly reevaluate community members so token-based roles are applied
+        self.owner.wakuext_service.reevaluate_community_members_permissions(community_id)
+
+        # Verify member is now in community and has admin role
+        communities = self.owner.wakuext_service.communities()
+        owner_community = next(
+            (c for c in self._communities_list(communities) if c.get("id") == community_id),
+            None,
+        )
+        assert owner_community is not None
+
+        member_key = self.member_with_snt.public_key
+        owner_key = self.owner.public_key
+
+        # Member should have admin role (4), granted via BECOME_ADMIN token permission
+        assert member_key in owner_community.get("members", {})
+        assert 4 in owner_community["members"][member_key].get("roles", [])
```

Comment:
Please define a `enum` with roles instead of hardcoding `4`.
https://github.com/status-im/status-go/blob/b12310d62cc9cca30d700be80b56f35bb6d695aa/protocol/protobuf/communities.proto#L19-L26

Reply:
Fixed at https://github.com/status-im/status-go/pull/7113/commits/6901bcccf8b2c4f0b24194bc30edb0e6dd8544e1